test/algorithms_test.dart

```diff
@@ -366,6 +366,15 @@ void main() {
     reverse(l, 0, 6);
     expect(l, equals([2, 1, 4, 3, 6, 5]));
   });
+
+  test('mergeSort works when runtime generic is a subtype of the static type',
+      () {
+    // Regression test for https://github.com/dart-lang/collection/issues/317
+    final length = 32; // _mergeSortLimit
```

Comment:
Should probably test above and below the limit - the limit selects a different path.

If the limit is changed, how does this git updated? Perhaps use a value significantly larger.

Reply:
Below the limit should be covered by other tests. This test only needs to ensure that the type error would occur. I'll bump it up to a larger number for safety, I don't think it's worth making the constant public for testing.